gcloud/bigquery/dataset.py

```diff
@@ -204,6 +273,28 @@ def _require_client(self, client):
             client = self._client
         return client
 
+    def _parse_access_grants(self, access):
+        """Parse a resource fragment into a schema field.
+
+        :type access: list of mappings
+        :param access: each mapping represents a single access grant
+
+        :rtype: list of :class:`AccessGrant`
+        :returns: a list of parsed grants
+        """
+        result = []
+        for grant in access:
+            role = grant['role']
+            if 'specialGroup' in grant:
+                entity_type = 'specialGroup'
+            elif 'groupByEmail' in grant:
+                entity_type = 'groupByEmail'
+            else:
```

Comment:
RE:

> parsing a resource returned by the back-end, not user-supplied data, so we shouldn't be raising exceptions at the user.

What do we do if the backend breaks the contract with us? Or if the server updates the list of enums and our code hasn't been updated yet (or someone just uses an old version of the library)?

If we don't raise an exception, how do we let the user know something has gone wrong / unexpected?